ixmp/backend/ixmp4.py

```diff
@@ -0,0 +1,91 @@
+from typing import TYPE_CHECKING
+
+from ixmp.backend.base import CachingBackend
+
+if TYPE_CHECKING:
+    import ixmp4
+
+
+class IXMP4Backend(CachingBackend):
+    """Backend using :mod:`ixmp4`."""
+
+    _platform: "ixmp4.Platform"
+
+    def __init__(self):
+        import ixmp4
+
+        # TODO Obtain `name` from the ixmp.Platform creating this Backend
+        name = "test"
+
+        # Add an ixmp4.Platform using ixmp4's own configuration code
+        # TODO Move this to a test fixture
+        # NB ixmp.tests.conftest.test_sqlite_mp exists, but is not importable (missing
+        #    __init__.py)
+        import ixmp4.conf
+
+        dsn = "sqlite:///:memory:"
+        try:
+            ixmp4.conf.settings.toml.add_platform(name, dsn)
+        except ixmp4.core.exceptions.PlatformNotUnique:
+            pass
+
+        # Instantiate and store
+        self._platform = ixmp4.Platform(name)
+
+    def get_scenarios(self, default, model, scenario):
+        # Current fails with:
+        # sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: run
+        # [SQL: SELECT DISTINCT run.model__id, run.scenario__id, run.version,
+        # run.is_default, run.id
+        # FROM run
+        # WHERE run.is_default = 1 ORDER BY run.id ASC]
+        return self._platform.runs.list()
+
+    # The below methods of base.Backend are not yet implemented
```

Comment:
FYI @glatterf42, the idea is that to remove lines like `name = _ni` below as each method gets a concrete implementation, in parallel with tests that exercise it.